client/js/chart/index.js

```diff
@@ -0,0 +1,131 @@
+import {SVG_NS} from '../util.js';
+
+/**
+ * Creates an SVG chart inside of `el` with the given data.
+ *
+ * @param {HTMLElement} el The container element for this chart.
+ * @param {Map<string, number[]>} rankingHistory The ranking history for each
+ * track.
+ * @param {Date[]} rankingDates The date of each history entry.
+ */
+export function createChart(el, rankingHistory, rankingDates) {
+  const svg = document.createElementNS(SVG_NS, 'svg');
+  svg.setAttribute('class', 'chart');
+  svg.append(createDefs());
+
+  const seriesContainer = document.createElementNS(SVG_NS, 'g');
+  seriesContainer.setAttribute('class', 'series-container');
+  svg.append(seriesContainer);
+
+  let index = 0;
+  const colors = ['blue', 'red', 'green', 'purple', 'orange'];
+
+  // get only the first 7 tracks for now
+  for (const history of [...rankingHistory.values()].slice(0, 7)) {
```

Comment:
What is the `...` notation mean in `[...rankingHistory]`?

Reply:
[Spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)